core/src/main/java/com/homeaway/streamplatform/streamregistry/StreamRegistryApplication.java

```diff
@@ -390,4 +406,49 @@ private void registerManagedContainer(final Environment environment, ManagedKStr
+        try {
+            TopicDescription topicDescription = describeTopicsResult.values().get(topicName).get();
+            log.debug("EventStore topic {} present. Topic Details = {}", topicName, topicDescription);
+            return true;
+        } catch (InterruptedException | ExecutionException e) {
+            // If the exception is an instance of UnknownTopicOrPartitionException, then the topic is not available.
+            if (e.getCause() instanceof UnknownTopicOrPartitionException) {
+                log.error("EventStore topic {} not present.", topicName);
+                return false;
+            }
+            /* If exception is not an instance of UnknownTopicOrPartitionException, then the Kafka Cluster is not reachable,
+             and it is better to halt the application by throwing back the exception. */
+            throw new RuntimeException(String.format("Error while describing the EventStore topic=%s. " +
+                            "Please make sure the cluster is accessible" , topicName), e);
+        }
+    }
+
+    protected void validateTopicConfigs(AdminClient adminClient, EventStoreTopic eventStoreTopic) throws ExecutionException, InterruptedException {
+        ConfigResource configResource = new ConfigResource(ConfigResource.Type.TOPIC, eventStoreTopic.getName());
+        // check whether the compaction property is available
+        Config config = adminClient.describeConfigs(Collections.singleton(configResource)).values().get(configResource).get();
+
+        Map<String, String> expectedProperties = eventStoreTopic.getProperties();
+        Collection<ConfigEntry> actualProperties = config.entries();
+
+        expectedProperties.forEach((key, value) -> {
+            boolean isConfigAvailable = actualProperties.stream().anyMatch(configEntry ->
+                    configEntry.name().equals(key) && configEntry.value().equals(value));
+            if (! isConfigAvailable)
+                throw new RuntimeException(String.format("EventStore topic config mismatch. TopicName=%s Excepted Config=%s ;" +
+                    " Actual Config=%s", eventStoreTopic.getName(), expectedProperties, actualProperties));
+        });
+        log.info("Stream Registry underlying EventStore kafka topic config {} validation passed", actualProperties);
+    }
+
+    private void createTopicWithCompactionEnabled(AdminClient adminClient, EventStoreTopic eventStoreTopic) throws ExecutionException, InterruptedException {
+        NewTopic streamRegistryTopic = new NewTopic(eventStoreTopic.getName(), eventStoreTopic.getPartitions(), eventStoreTopic.getReplicationFactor());
+        streamRegistryTopic.configs(eventStoreTopic.getProperties());
+        adminClient.createTopics(Collections.singleton(streamRegistryTopic)).values().get(eventStoreTopic.getName()).get();
+        log.info("Stream Registry underlying kafka event store created : {}", streamRegistryTopic);
+    }
```

Comment:
All three of these methods are more static utility methods.

1. They are specific to Kafka. 
2. They don't need to be in the Dropwizard Application class.

Reply:
Good callout @cricket007.

Should this be part of a configure method in `InfraManager` ?

Reply:
Moved the Kafka Client calls to a Utils

Reply:
Maybe, though, even then the healthcheck of the entire app will also now be coupled with Kafka, and not "disableable".